lib/server-config.ts

```diff
@@ -36,6 +38,19 @@ export const parseServerConfig = (
       canSendEmail: !!config.pds?.url && isModerator,
       canManageTeam: isAdmin,
       canTakedownFeedGenerators: isAdmin,
+      canDivertBlob: !!config.blobDivert?.url && isModerator,
     },
   }
 }
+
+export const checkPermission = (
+  permission: keyof ServerConfig['permissions'],
+) => {
+  try {
+    return client.session.serverConfig?.permissions[permission]
+  } catch (e) {
+    // Trying to access client.session while unauthenticated throws an error in which case,
+    // we can safely assume the user does not have the permission
+    return false
+  }
```

Comment:
This works, but could also rely on `client.authState` to avoid leaning on throwing:
```ts
if (client.authState === AuthState.LoggedOut) {
  return false
}
return client.session.serverConfig?.permissions[permission]
```

Reply:
hmm... I know these are directly linked but feels a bit odd just within this context to assume that logged in state guarantees existnence of session. I'll leave it as if that's ok?